src/holofuel/pages/Inbox/Inbox.js

```
@@ -58,31 +68,122 @@ function useFetchCounterparties () {
+  const { data: { holofuelUser: whoami = {} } = {} } = useQuery(HolofuelUserQuery)
+  const { data: { holofuelLedger: { balance: holofuelBalance } = { balance: 0 } } = {} } = useQuery(HolofuelLedgerQuery)
+  const { data: { holofuelActionableTransactions: actionableTransactions = [] } = {} } = useQuery(HolofuelActionableTransactionsQuery)
+  const { data: { holofuelRecentTransactions: recentTransactions = [] } = {} } = useQuery(HolofuelRecentTransactionsQuery)
 
   useFetchCounterparties()
   const payTransaction = useOffer()
   const declineTransaction = useDecline()
-  const [modalTransaction, setModalTransaction] = useState()
-  const isTransactionsEmpty = isEmpty(transactions)
-
-  const pageTitle = `Inbox${isTransactionsEmpty ? '' : ` (${transactions.length})`}`
+  const [toggleModal, setToggleModal] = useState(null)
+  const [modalTransaction, setModalTransaction] = useState(null)
 
-  const showConfirmationModal = (transaction, action) => {
+  const showConfirmationModal = (transaction = {}, action = '') => {
     const modalTransaction = { ...transaction, action }
-    setModalTransaction(modalTransaction)
+    if (!isEmpty(transaction) && action !== '') return setModalTransaction(modalTransaction)
+    else return setToggleModal(true)
   }
 
-  return <PrimaryLayout headerProps={{ title: pageTitle }}>
+  const [actionsVisible, setActionsVisible] = useState(null)
+  const actionsClickWithTx = transaction => setActionsVisible(transaction)
+
+  const toggleButtons = [{ view: 'pending', label: 'Pending' }, { view: 'recent', label: 'Recent' }]
+  const [inboxView, setInboxView] = useState(VIEW.pending)
+  let displayTransactions = []
+  switch (inboxView) {
+    case VIEW.pending:
+      displayTransactions = actionableTransactions
+      break
+    case VIEW.recent:
+      displayTransactions = recentTransactions
+      break
+    default:
+      displayTransactions = actionableTransactions
```

I'd throw an error here. If we hit this clause, something has gone wrong.